modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -410,7 +410,7 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
          *  @param outputName name for layer which output is needed to get
          *  @details If @p outputName is empty, runs forward pass for the whole network.
          */
-        CV_WRAP void forward(OutputArrayOfArrays outputBlobs, const String& outputName = String());
+        CV_WRAP void forward(CV_OUT OutputArrayOfArrays outputBlobs, const String& outputName = String());
```

Comment:
@arrybn, thank you! Python bindings generator can do it by `OutputArrayOfArrays`: https://docs.opencv.org/master/da/d49/tutorial_py_bindings_basics.html

> Header parser can understand the input and output arguments from keywords like InputArray, OutputArray etc. But sometimes, we may need to hardcode inputs and outputs. For that, macros like CV_OUT, CV_IN_OUT etc. are used.

Reply:
Agree with @dkurt . CV_OUT is not necessary here.
Perhaps there is some bug/issue with bindings generator itself.
Looks like nothing was changed by this patch in Python signatures:
- [master](https://docs.opencv.org/master/db/d30/classcv_1_1dnn_1_1Net.html#a8d248f50efec70dffa785de18a4b2170)
- [patch](http://pullrequest.opencv.org/buildbot/export/pr/10331/docs/db/d30/classcv_1_1dnn_1_1Net.html#a8d248f50efec70dffa785de18a4b2170)

Workaround is to pass None:
```
outputBlobs = new.forward(outputBlobs=None, outputName='name')
```

BTW, There is this method: `CV_WRAP Mat forward(const String& outputName = String());`. Python binding of this method has strong conflict with method above (can't be resolved properly without renaming like `CV_WRAP_AS`).